localstack/services/opensearch/versions.py

```diff
@@ -220,6 +221,7 @@
         TargetVersions=["OpenSearch_2.5"],
     ),
     CompatibleVersionsMap(SourceVersion="OpenSearch_2.5", TargetVersions=["OpenSearch_2.7"]),
+    CompatibleVersionsMap(SourceVersion="OpenSearch_2.7", TargetVersions=["OpenSearch_2.9"]),
```

Comment:
This is incomplete, you will also need to update the `OpenSearch_2.X` version maps since that offers a valid upgrade path to 2.9.

Refer to AWS's `GetCompatibleVersions` operation for full map. You will also need to update the `test_get_compatible_versions` test.